test/e2e/flask/multichain-api/wallet_invokeMethod.spec.ts

```diff
@@ -172,77 +183,63 @@ describe('Multichain API', function () {
             ...DEFAULT_MULTICHAIN_TEST_DAPP_FIXTURE_OPTIONS,
           },
           async ({ driver, extensionId }: FixtureCallbackArgs) => {
-            await unlockWallet(driver);
+            await loginWithBalanceValidation(driver);
 
             const testDapp = new TestDappMultichain(driver);
             await testDapp.openTestDappPage();
+            await testDapp.check_pageIsLoaded();
             await testDapp.connectExternallyConnectable(extensionId);
             await testDapp.initCreateSessionScopes(
               GANACHE_SCOPES,
               CAIP_ACCOUNT_IDS,
             );
-            await addAccountInWalletAndAuthorize(driver);
-            await driver.clickElementAndWaitForWindowToClose({
-              text: 'Connect',
-              tag: 'button',
-            });
+            const connectAccountConfirmation = new ConnectAccountConfirmation(
+              driver,
+            );
+            await connectAccountConfirmation.check_pageIsLoaded();
+            await connectAccountConfirmation.confirmConnect();
+
             await driver.switchToWindowWithTitle(
               WINDOW_TITLES.MultichainTestDApp,
             );
-
-            for (const [i, scope] of GANACHE_SCOPES.entries()) {
-              await driver.clickElementSafe(
-                `[data-testid="${replaceColon(
-                  scope,
-                )}-eth_sendTransaction-option"]`,
-              );
-
-              i === INDEX_FOR_ALTERNATE_ACCOUNT &&
-                (await driver.clickElementSafe(
-                  `[data-testid="${replaceColon(scope)}-${ACCOUNT_2}-option"]`,
-                ));
+            await testDapp.check_pageIsLoaded();
+            for (const scope of GANACHE_SCOPES) {
+              await testDapp.selectMethod({
+                scope,
+                method: 'eth_sendTransaction',
+              });
             }
 
-            await driver.clickElement({
-              text: 'Invoke All Selected Methods',
-              tag: 'button',
-            });
-
+            await testDapp.clickInvokeAllMethodsButton();
             const totalNumberOfScopes = GANACHE_SCOPES.length;
             for (let i = 0; i < totalNumberOfScopes; i++) {
-              await driver.delay(largeDelayMs);
               await driver.switchToWindowWithTitle(WINDOW_TITLES.Dialog);
-              await driver.clickElement({
-                text: 'Confirm',
-                tag: 'button',
-              });
+              const confirmation = new Confirmation(driver);
+              await confirmation.check_pageIsLoaded();
+              await confirmation.clickFooterConfirmButton();
             }
 
-            await driver.delay(largeDelayMs);
             await driver.switchToWindowWithTitle(
```

Comment:
Fix another flaky test:
`await driver.delay(largeDelayMs);` is not long enough sometimes.

The fix is to replace the hard delay with a dynamic check:
First, verify on the homepage that the transaction has been completed.
Then, proceed to check the account balance in the next step.